modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/DiscoverAdapter.kt

```diff
@@ -785,7 +810,11 @@ internal class DiscoverAdapter(
                 categoriesViewHolder.binding.lblTitle.contentDescription = tittle
             }
             categoriesViewHolder.adapter.fromListId = row.listId
-            categoriesViewHolder.adapter.submitList(row.podcasts) { onRestoreInstanceState(categoriesViewHolder) }
+            categoriesViewHolder.adapter.submitList(null)
+            categoriesViewHolder.adapter.submitList(row.podcasts)
```

Comment:
Instead of doing this it would be good to add a separate method to the adapter like `replaceList()` that would do this. It'd be nice to add a comment that we need to do this because otherwise there are delays in displaying categories that are a result of diffing computations.

Reply:
I agree! It's done here: [45f7eda](https://github.com/Automattic/pocket-casts-android/pull/1975/commits/45f7eda17b1b0ee29c9870a621d2add38a965ca9)